bin/program.py

```diff
@@ -145,7 +145,7 @@ def __init__(
                 problem.path.resolve() / self.subdir
             )
             self.short_path = relpath
-            self.name = str(relpath)
+            self.name: str = str(relpath)
```

Comment:
I don't think an explicit type annotation is needed here because `str` deterministically returns `str`, so that can always be inferred, right?

Reply:
we probably should annotate all variables, i.e. make clear that `str` is intended and not `Optional[str]` but i did the bare minimum outside of util

Reply:
> we probably should annotate all variables

We don't _have_ to do this :slightly_smiling_face: We do want to enable the `--strict` flag at some point (see https://github.com/RagnarGrootKoerkamp/BAPCtools/blob/master/.pre-commit-config.yaml#L33), but that only requires function signatures to be fully type-annotated. For fields and variables, MyPy can infer the type from the assigned value, which does not change anymore after its initial assignment. So in this case, `self.name = str(relpath)` will lock the type of `self.name` to `str`, and trying to assign `None` to it later, will throw an error :smile:

Reply:
Yeah my point here is that I want to pin those attribute types. Even though we don't yet reassign those variables we might want to do that in the future and specifying that we really want a `str` here and not an `Optional[str]` makes this more precise then the automatically interfered `str`.

Reply:
Fair enough! In that case, we may want to find some automated tooling that enforces type annotations for class fields, but let's leave that for another time (I'll add it to #102) :slightly_smiling_face: